api/src/routes/auth/validation.ts

```diff
@@ -1,4 +1,43 @@
 import { z } from 'zod';
+import { AccessLevelSchema } from '../../utils/auth.ts';
+import { IdAndSchemaVersionSchema } from '../../utils/db.ts';
+
+export const ProfileStatusSchema = z.enum([
+	'activated',
+	'created',
+	'deleted',
+	'error',
+	'profile-completed',
+	'social-handle-provided',
+	'tos-accepted'
+]).describe('The status a profile may be in. It is useful for the sign-up process.');
+export type ProfileStatus = z.infer<typeof ProfileStatusSchema>;
+
+export const AccessSchema = IdAndSchemaVersionSchema.merge(z.object({
+	accessLevel: AccessLevelSchema,
+	password: z
+		.string()
+		.describe("The user's password, hashed and salted."),
+	email: z
+		.string()
```

Comment:
suggestion: we should check if the email is a valid email
```suggestion
		.string()
		.email()
```